docs/io/fits/index.rst

```diff
@@ -589,6 +589,42 @@ lines to accomplish the same behavior::
 This will write a single HDU to a FITS file without having to manually
 encapsulate it in an :class:`HDUList` object first.
 
+Creating an Image File with Multiple Extensions
+"""""""""""""""""""""""""""""""""""""""""""""""
+
+In the previous example we created an :class:`HDUList` object with a single
+extension (a :class:`PrimaryHDU`). To create a file with multiple extensions we need to
+use :class:`ImageHDU`\s and append them to an :class:`HDUList`.
+
+First, we create some more data::
+
+    >>> n2 = np.ones((3, 3))
+    >>> n3 = np.ones((100, 100))
+
+Note that the data shapes of the different extensions do not need to be the same. Next,
+place the data into separate :class:`ImageHDU` objects::
+
+    >>> hdu2 = fits.ImageHDU(n2)
+    >>> hdu3 = fits.ImageHDU(n3)
```

Comment:
It would be useful to mention here the possibility to give a name to HDUs. Also the variable names starting at 2 look a bit weird, why not using a more real word like example ? Suggestion:
```
    >>> data_hdu = fits.ImageHDU(data=np.zeros(10), name='DATA')
    >>> var_hdu = fits.ImageHDU(data=np.ones(10), name='VAR')
    >>> hdul = fits.HDUList([data_hdu, var_hdu])
```
And then with 'DQ' for the third extension ?

Reply:
I'm going to respectfully push back on this. The additions in this PR are designed to follow naturally from the previous section that details creating a single-HDU file with a ``PrimaryHDU``. In other words, I view this new subsection as an extension of its parent section and as such I think the form should be the same.

The point here is that it's almost trivially easy to add another HDU with nothing more than a numpy array. I'd worry that throwing `data` and `name` keywords will make this look more complicated than the ``PrimaryHDU`` case. The finer details you mentioned are still only a click a way for those more curious than the basic example.

Reply:
Actually after reading the whole section, I think that this should be a new subsection after the "Creating a New Table File" and at the same level. Because multi-extension files can contain table as well I would rename to "Creating a file with Multiple Extensions", and maybe include a table extension. Also I think that the variable names should make sense even when you don't read the previous sections.

Reply:
That's a good point, @saimn, the New Table section already has some of this information in it, and because an HDUList can have any type of HDU in it it makes sense to make this a separate section (not a subsection of the image section). I've moved the next section to be after the New Table section and consolidated the information.